library/common-library/DEVELOPERS.md

```diff
@@ -0,0 +1,615 @@
+
+Usage:
+```mustache
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  labels:
+    {{- include "newrelic.common.labels" . | nindent 4 }}
+  name: {{ include newrelic.common.naming.truncateToDNSWithSuffix (dict "name" (include "newrelic.common.naming.fullname" .) suffix "agent-config") }}
+  namespace: {{ .Release.Namespace }}
+data:
+  newrelic-infra.yml: |-
+    # This is the configuration file for the infrastructure agent. See:
+    # https://docs.newrelic.com/docs/infrastructure/install-infrastructure-agent/configuration/infrastructure-agent-configuration-settings/
+    {{- include "newrelic.common.agentConfig.defaults" . | nindent 4 }}
+```
+
+
+
+## _cluster.tpl
+### `newrelic.common.cluster`
+Returns the cluster name
+
+Usage:
+```mustache
+{{ include "newrelic.common.cluster" . }}
+```
+
+
+
+## _custom-attributes.tpl
+### `newrelic.common.customAttributes`
+Return custom attributes in YAML format.
+
+Usage:
+```mustache
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: example
+data:
+  custom-attributes.yaml: |
+    {{- include "newrelic.common.customAttributes" . | nindent 4 }}
+  custom-attributes.json: |
+    {{- include "newrelic.common.customAttributes" . | fromYaml | toJson | nindent 4 }}
+```
+
+
+
+## _fedramp.tpl
+### `newrelic.common.fedramp.enabled`
+Returns true if FedRAMP is enabled or an empty string if not. It can be safely used in conditionals as an empty string is a Helm falsiness.
+
+Usage:
+```mustache
+{{ include "newrelic.common.fedramp.enabled" . }}
+```
+
+### `newrelic.common.fedramp.enabled.value`
+Returns true if FedRAMP is enabled or false if not. This is to have the value of FedRAMP ready to be templated.
+
+Usage:
+```mustache
+{{ include "newrelic.common.fedramp.enabled.value" . }}
+```
+
+
+
+## _license.tpl
+### `newrelic.common.license.secretName` and ### `newrelic.common.license.secretKeyName`
+Returns the secret and key inside the secret where to read the license key.
+
+The common library will take care of using a user-provided custom secret or creating a secret that contains the license key.
+
+To create the secret use `newrelic.common.license.secret`.
+
+Usage:
+```mustache
+{{- if and (.Values.controlPlane.enabled) (not (include "newrelic.fargate" .)) }}
+apiVersion: v1
+kind: Pod
+metadata:
+  name: example
+spec:
+  containers:
+  - name: agent
+    env:
+    - name: "NRIA_LICENSE_KEY"
+      valueFrom:
+        secretKeyRef:
+          name: {{ include "newrelic.common.license.secretName" . }}
+          key: {{ include "newrelic.common.license.secretKeyName" . }}
+```
+
+
+
+## _license_secret.tpl
+### `newrelic.common.license.secret`
+This function templates the secret that is used by agents and integrations with the license Key provided by the user. It will
+template nothing (empty string) if the user provides a custom pair of secret name and key.
+
+This template also does the safety checks of failing in case the user has not provided any license key or custom secret.
```

Comment:
```suggestion
This template fails in case the user has not provided any license key or custom secret.
```

Reply:
I rephrased it as I wanted to state that there is no need to make safety checks. The common library does that for you.

Reply:
Fixed in [#6216d9f](https://github.com/newrelic/helm-charts/pull/797/commits/6216d9f3f09fedf3717c1b7a457a887fd8230371)